### PR TITLE
added power cone to cones.rst

### LINKED ISSUE #70: Update conic interface to add power cones
SCS just added power cones `{(x,y,z) | x^a * y^(1-a) >= |z|, x>=0, y>=0}`! Let's add these to the MathProgBase conic interface. (I think this only requires a change to the docs.)

### PR DESCRIPTION
**Pull request by madeleineudell**
closes #70

### CODE REVIEW

**Comment by joehuchette** [comment]
:+1:

File: doc/conic.rst [hunk 1]
```diff
@@ -36,6 +36,7 @@ The recognized cones are:
 - ``:SDP``, the cone of symmetric positive semidefinite matrices :math:`\{ X \in \mathbb{R}^{n\times n} : X \succeq 0\}`
 - ``:ExpPrimal``, the exponential cone :math:`\operatorname{cl}\{ (x,y,z) \in \mathbb{R}^3 : y > 0, y e^{x/y} \leq z \}`
 - ``:ExpDual``, the dual of the exponential cone :math:`\{ (u,v,w) \in \mathbb{R}^3 : u < 0, -ue^{v/q} \leq ew\} \cup \{(0,v,w) : v \geq 0, w \geq 0\}` 
+- ``:Power``, the power cone :math:`\{ (x,y,z) \in \mathbb{R}^3 : x^a y^(1-a) \geq |z|, x\geq0, y\geq0\}`
```
Comment (mlubin) [inline doc/conic.rst]: I think you need a space after ``geq`` for the formatting: ``x \geq 0``

**Comment by mlubin** [comment]
The definition of the power cone [here](http://www.seas.ucla.edu/~vandenbe/236C/lectures/conic.pdf) has an upper bound with a product of an arbitrary number of products. Is the definition here with a product of two elements reasonably standard, or is it just what happens to be implemented in SCS?
@bodono

**Comment by bodono** [comment]
We can just consider the three-dimensional cone, because you can construct larger power cones by a product of 3 dimensional power cones by adding new variables, and most applications that I know of need only the 3d power cone (such as the conic representation of p-norms).

Here are some other papers that restrict consideration to the 3-dimensional cone:
[Nesterov, top of section 7](http://www.optimization-online.org/DB_FILE/2006/03/1355.pdf), [Skajaa and Ye, section 6.1](http://web.stanford.edu/~yyye/nonsymmhsdimp.pdf, section 6.1)

**Comment by mlubin** [comment]
Fair enough. We also have an issue here in that cone itself has a parameter ``a`` which the current API doesn't make it clear how to provide.

**Comment by madeleineudell** [comment]
Suggestions for how to encode the parameter `a`? For example, we could write `a` via an additional variable: `(x,y,z,a) \in K <=> x^a y^(1-a) \geq |z|, x\geq 0, y\geq 0`. Then we'd encode the cone by specifying a block of **4** rows of `b` and `A`, with the fourth row of `b` representing `a`.

**Comment by mlubin** [comment]
Using a fake variable to define a cone is a bit icky. What about defining the cone with a tuple ``(:Power, a)`` instead of the symbol? This breaks code which assumes that the cone types are symbols, but I don't think that's a big deal.

**Comment by IainNZ** [comment]
We could just use immutable types defined in MPB actually at this point, e.g.

```julia
abstract AbstractCone

type NonNegCone <: AbstractCone
  idx::Vector{Int}
end

type PowerCone <: AbstractCone
  idx::Vector{Int}
  a
end
```

**Comment by madeleineudell** [comment]
Oh yes, the immutable types version is much nicer!

**Comment by mlubin** [comment]
That's a bit more breaking but okay also. What about just using a tuple of ``(a,idx)`` in the indices field? We probably wouldn't want to force the ``idx`` field to be ``Vector{Int}`` since we currently allow ranges instead.

**Comment by mlubin** [comment]
@chriscoey, we could revive this PR

**Comment by chriscoey** [comment]
oh let's!

**Comment by chriscoey** [comment]
Let's fix this. Then we can ask SCS to support adding power cones through the conic interface (probably not a big change). Then hopefully Convex.jl can support power cone problems too (bigger change). @mlubin @madeleineudell thoughts?

**Comment by madeleineudell** [comment]
That sounds great. We can definitely get power cones implemented in Convex after they're ready in MPB. cc @ramcha24

**Comment by chriscoey** [comment]
Great! maybe @mlubin and I can get to this during MIP conference in a couple weeks

once again i am left wishing github had a "set reminder" function

**Comment by madeleineudell** [comment]
boomerang for gmail...?

On Mon, May 9, 2016 at 2:27 PM, Chris C. <notifications@github.com> wrote:

> Great! maybe @mlubin <https://github.com/mlubin> and I can get to this
> during MIP conference in a couple weeks
>
> once again i am left wishing github had a "set reminder" function
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaOpt/MathProgBase.jl/pull/71#issuecomment-217995329>
>



-- 
Madeleine Udell
Postdoctoral Fellow at the Center for the Mathematics of Information
California Institute of Technology
*https://courses2.cit.cornell.edu/mru8
<https://courses2.cit.cornell.edu/mru8>*
(415) 729-4115

**Comment by chriscoey** [comment]
Yes good point. I just like to complain.

**Comment by chriscoey** [comment]
So is there a nice way to specify a power cone with a given parameter without breaking MathProgBase?

**Comment by chriscoey** [comment]
Also, @madeleineudell, I want to do a little cost-benefit analysis for whether power cones are worth implementing this Summer, because @mlubin thinks they should be low priority. To do that I need to know whether there are important example problems. So we know that rational powers can be reformulated with SOCP, but potentially lots of SOCs are needed. Do you have any ideas for problems where it's really significantly better to use a power cone than an SOCP reformulation?

**Comment by mlubin** [comment]
Closing in favor of https://github.com/JuliaOpt/MathProgBase.jl/issues/168

**Pull request closed without merging**